src/Codeception/Subscriber/FailFast.php

```diff
@@ -16,12 +17,33 @@ class FailFast implements EventSubscriberInterface
      * @var array<string, string>
      */
     protected static array $events = [
-        Events::SUITE_BEFORE => 'stopOnFail',
+        Events::TEST_FAIL => 'stopOnFail',
+        Events::TEST_ERROR => 'stopOnFail',
+        Events::SUITE_BEFORE => 'cacheSuite'
     ];
 
-    public function stopOnFail(SuiteEvent $event): void
+    private $failureCount = 0;
+
+    private $stopFailureCount;
+
+    private $suiteCache;
+
+    public function __construct($stopFailureCount)
+    {
+        $this->stopFailureCount = (int) $stopFailureCount;
+    }
```

Comment:
how about typing `$stopFailureCount` parameter and adding the `int` cast in src/Codeception/Codecept.php:L132 ?

Reply:
Thanks, I implemented all changes.